src/aiotk/posix.py

```diff
@@ -115,20 +135,28 @@ def close(self):
     # - not clear if we can still accept connections until
     #   ```self._server.wait_closed()` completes, so we cancel again after it
     #   does.
-    async def wait_closed(self, timeout=None):
+    async def wait_closed(self, timeout: Optional[float]=None) -> None:
         """Wait until all connections are closed.
 
         See:
 
         - :py:meth:`aiotk.TCPServer.wait_started`
         - :py:meth:`aiotk.TCPServer.wait_closed`
         """
+        # NOTE: there is a natural race here and we could not manage to
+        #       reliably reach this branch while the boot task had not yet
+        #       completed.
+        if self._boot and not self._boot.done():
+            raise Exception('Starting.')  # pragma: no cover
         if not self._server:
             return
         await self._server.wait_closed()
         for session in self._sessions:
             session.cancel()
-        pending = self._sessions
+        # NOTE: `mypy` does not recognize `Set[Task[Any]]` as a subtype of
+        #       `Set[Future[Any]]`.  The only way to resolve this is to create
+        #       a copy...
+        pending = {t for t in self._sessions}  # type: Set[Future[Any]]
```

Comment:
**Note to self**: report this feedback to `mypy` devs to ensure they are aware of this limitation in `mypy`.

Reply:
Tried running a simplified example and could not reproduce.  Guess I will have to experiment some more :-(